lib/AST/Decl.cpp

```diff
@@ -2906,6 +2906,18 @@ bool ValueDecl::isDynamic() const {
     getAttrs().hasAttribute<DynamicAttr>());
 }
 
+bool ValueDecl::isObjCDynamicInGenericClass() const {
```

Comment:
This new predicate is called in a number of places and it looks like it might be easy to miss one. Can we instead change isNativeDynamic() and isObjCDynamic() to do the right thing here, and treat a generic class as if it were not "really" ObjC?

Reply:
No, there are places where you want to return isObjcDynamic() == true and isObjCDynamicInGenericClass() is true.

You still want to treat calls to objc methods in generic classes as objc methods.

Reply:
Perhaps we can separate the predicate into 'how to call' and 'how to codegen' behaviors?

Reply:
I will try that